src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs

```diff
@@ -56,7 +56,14 @@ internal sealed class SolutionRestoreCommand
             var menuCommandId = new CommandID(CommandSet, CommandId);
             var menuItem = new OleMenuCommand(
                 OnRestorePackages, null, BeforeQueryStatusForPackageRestore, menuCommandId);
-            commandService?.AddCommand(menuItem);
+
+            // call AddCommand through explicitly moving to UI thread since this is now being 
+            // initiliazed as part of AsynPackage
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
What about the other statement below ` _vsMonitorSelection.GetCmdUIContextCookie`? Should this also be moved to UI thread?